---
title: C26111 |Microsoft Docs
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
caps.latest.revision: 10
author: corob-msft
ms.author: corob
manager: jillfra
ms.openlocfilehash: d8a9adb03db78b5d6c6dda113ba91767a02d06e6
ms.sourcegitcommit: 6cfffa72af599a9d667249caaaa411bb28ea69fd
ms.translationtype: MT
ms.contentlocale: ja-JP
ms.lasthandoff: 09/02/2020
ms.locfileid: "77278829"
---
# <a name="c26111"></a>C26111
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

警告 C26111: 呼び出し元が関数を呼び出す前にロックを解放でき \<lock> \<func> ませんでした。  
  
 注釈は、 `_Requires_lock_not_held_` 関数が呼び出されたときに、指定されたロックのロックカウントが0を超えることができないことを前提としています。 警告 C26111 は、関数が別の関数を呼び出す前にロックの解放に失敗したときに発行されます。  
  
## <a name="example"></a>例  
 次の例では、 `_Requires_lock_not_held_` ロックされたセクション内のへの呼び出しによって事前条件に違反するため、警告 C26111 が生成され `DoNotLock` ます。  
  
```  
  
typedef struct _DATA   
{  
    CRITICAL_SECTION cs;  
    int d;  
} DATA;  
  
_Requires_lock_not_held_(p->cs)   
  
void DoNotLock(DATA* p)  
{  
    EnterCriticalSection(&p->cs);   
    p->d = 0;  
    LeaveCriticalSection(&p->cs);  
}  
  
void LockedFunction(DATA* p)   
{   
    EnterCriticalSection(&p->cs);   
    DoNotLock(p); // Warning C26111  
    LeaveCriticalSection(&p->cs);  
}  
  
```
